### DISCUSSION TITLE
About BasicAuth

### DISCUSSION THREAD

**Discussion by josejachuf** [salvo-rs/salvo#488] [Q&A] [answered]
I'm trying to add basic_auth_handler to some routes, that are in different branches of the route tree

let basic_auth_handler = BasicAuth::new(auth::Validator);

The problem is that it moves in the first hoop I use. I tried to pass as a reference but does not allow it. Nor does the Trait Clone support.

```
.hoop(&basic_auth_handler)
    |                  ---- ^^^^^^^^^^^^^^^^^^^ the trait `Handler` is not implemented for `&BasicAuth<Validator>`

```
How can I do it?

**Comment by chrislearn** [comment, marked as answer]
you can call a function and let it create basic_auth for you.

**Reply by josejachuf** [reply]
This works fine!
Thanks
